pytorch_lightning/loops/training_batch_loop.py

```diff
@@ -255,6 +284,8 @@ def on_after_backward(self, batch_idx: int, untouched_loss: Tensor) -> None:
         # insert after step hook
         self.trainer.call_hook("on_after_backward")
 
+        self.progress.optimizations[opt_idx].optimizer.increment_ready()
```

Comment:
@tchaton do you have a recommendation of how the usage pattern should look like here with `opt_idx`, `batch_idx` etc.? 

Should we (or the user) pass down opt_idx to the functions as arguments or is the new pattern to rely on `self.progress.opt_idx` everywhere?

Also opinions @carmocca? 

I think it's important to figure out a consistency here because we don't want to question ourselves if opt_idx is really the same value as self.progress.opt_idx if both are there :)

Reply:
The above discussion is constrained to the loops and internal use of course, not talking about opt_idx in hooks on LM etc.

Reply:
I think I prefer relying on a shared `opt_idx` attribute. Could live in the loop itself or the loop progress

Reply:
Removed entirely to fit to Carlos API for now. We will see if we open up optimization again..